nfc/doc/type_2_tag.rst

```diff
@@ -0,0 +1,398 @@
+       The NFC Forum Device shall ignore this.
+   * - Lock Control TLV
+     - 0x01
+     - Defines details of the lock bits.
+   * - Memory Control TLV
+     - 0x02
+     - Identifies reserved memory areas.
+   * - NDEF Message TLV
+     - 0x03
+     - Contains an NDEF message.
+   * - Proprietary TLV
+     - 0xFD
+     - Contains tag proprietary information.
+   * - Terminator TLV
+     - 0xFE
+     - Contains the last TLV block in the data area.
+
+To write data to the tag, use the Type 2 Tag library functions :cpp:func:`nfc_t2t_payload_set` or :cpp:func:`nfc_t2t_payload_raw_set`.
+:cpp:func:`nfc_t2t_payload_set` configures a single NDEF TLV block based on a user-provided NDEF message.
+:cpp:func:`nfc_t2t_payload_raw_set` does not configure a TLV block, but the provided data must be organized in a TLV structure.
+
+Dynamic Lock and Reserved bytes
+===============================
+
+15 Dynamic Lock bytes are located after the data area. Each bit defines access conditions of 8 consecutive bytes of the tag data area, except for the first 48 bytes, whose access conditions are defined by `Static Lock bytes`_.
+The Dynamic Lock bits are set to 1 to indicate that the tag is read-only.
+
+The lock bytes are followed by 1 reserved byte (Rsvd) to get a multiple of 8 bytes.
+
+.. _t2t_command_set:
+
+Command set
+***********
+
+The current version of the Type 2 Tag library supports only one Type 2 Tag command type: the READ command.
+When a READ command is received, the tag responds with the data that is stored for the tag.
+
+READ command format
+===================
+
+The READ command has the following format:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Byte number
+     - Description
+     - Value
+   * - 1
+     - Code
+     - 0x30
+   * - 2
+     - Number of the 4-byte data block to read from the tag data
+     - 0x00 - 0xFF
+
+The library can send either a READ response (success) or a NACK response (failure):
+
+* A READ response contains the content of 4 data blocks starting with the requested data block (16 bytes).
+* A NACK response contains the error code 0x0, 0x1, 0x4, or 0x5 (4 bits).
+
+.. _t2t_programming:
+
+Programming a tag
+*****************
+
+To program a tag, complete the following steps:
+
+1. Implement a callback function that handles events from the Type 2 Tag library and register it:
+
+   .. code-block:: c
+
+      int err;
+      /* Callback for NFC events */
+      static void nfc_callback(void * context,
+                               enum nfc_t4t_event event,
+                               const u8_t * data,
+                               size_t data_length,
+                               u32_t flags)
+      {
+      ...
+      }
+      /* Set up NFC and register application callback for NFC events. */
+      err = nfc_t2t_setup(nfc_callback, NULL);
+      if (err) {
+		printk("Cannot setup NFC T2T library!\n");
+		return err;
+      }
+
+   The NFCT interrupt handler runs at priority level APP_LOW.
```

Comment:
There is no _APP_LOW_ in NCS, there is a Kconfig symbol in NFC Platform in NCS. So, I think either we should remove this information from here, or just say it's configurable by the NFC Platform. I wouldn't link to the exact Kconfig setting as that is part of NCS and theoretically other users of this library can set their own IRQ priority in their own way, depending of the runtime environment.

Reply:
removed